sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy

```diff
@@ -96,222 +94,345 @@ class APISpec extends Specification {
+
+        for (HttpPipelinePolicy policy : policies) {
+            builder.addPolicy(policy)
         }
 
-        try {
-            blobStorageServiceURL = getGenericServiceURL(getGenericCreds("BLOB_STORAGE_"))
+        if (testMode == TestMode.RECORD) {
+            builder.addPolicy(interceptorManager.getRecordPolicy())
         }
-        catch (Exception e) {
+
+        return builder.credential(credential).buildBlobClient()
+    }
+
+    BlobClient getBlobClient(SharedKeyCredential credential, String endpoint, String blobName) {
+        BlobClientBuilder builder = new BlobClientBuilder()
+            .endpoint(endpoint)
+            .blobName(blobName)
+            .httpClient(getHttpClient())
+            .httpLogDetailLevel(HttpLogDetailLevel.BODY_AND_HEADERS)
+
+        if (testMode == TestMode.RECORD) {
+            builder.addPolicy(interceptorManager.getRecordPolicy())
         }
 
-        try {
-            premiumServiceURL = getGenericServiceURL(getGenericCreds("PREMIUM_STORAGE_"))
+        return builder.credential(credential).buildBlobClient()
+    }
+
+    BlobClient getBlobClient(String endpoint, SASTokenCredential credential) {
+        BlobClientBuilder builder = new BlobClientBuilder()
+            .endpoint(endpoint)
+            .httpClient(getHttpClient())
+            .httpLogDetailLevel(HttpLogDetailLevel.BODY_AND_HEADERS)
+
+        if (credential != null) {
+            builder.credential(credential)
         }
-        catch (Exception e) {
+
+        if (testMode == TestMode.RECORD) {
+            builder.addPolicy(interceptorManager.getRecordPolicy())
         }
+
+        return builder.buildBlobClient()
     }
 
-    def cleanupSpec() {
-        Assume.assumeTrue("The test only runs in Live mode.", getTestMode().equalsIgnoreCase(RECORD_MODE))
-        cleanupContainers()
+    private HttpClient getHttpClient() {
+        HttpClient client
+        if (testMode == TestMode.RECORD) {
+            client = HttpClient.createDefault().wiretap(true)
+        } else {
+            client = interceptorManager.getPlaybackClient()
+        }
+
+        if (Boolean.parseBoolean(ConfigurationManager.getConfiguration().get("AZURE_TEST_DEBUGGING"))) {
+            return client.proxy(PROXY_OPTIONS)
+        } else {
+            return client
+        }
     }
 
-    def setup() {
-        Assume.assumeTrue("The test only runs in Live mode.", getTestMode().equalsIgnoreCase(RECORD_MODE))
-        String containerName = generateContainerName()
+    private Supplier<ProxyOptions> PROXY_OPTIONS = new Supplier<ProxyOptions>() {
+        @Override
+        ProxyOptions get() {
+            return new ProxyOptions(ProxyOptions.Type.HTTP, new InetSocketAddress("localhost", 8888))
+        }
+    }
 
-        primaryServiceURL = getGenericServiceURL(primaryCreds)
-        cu = primaryServiceURL.getContainerClient(containerName)
-        cu.create()
+    def generateContainerName() {
```

Comment:
Is this still in use?

Reply:
Yes, it's used in 30 places based on IntelliJ's find all references.